Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs

```diff
@@ -216,11 +217,22 @@ void callOnFreeSlotClick()
 		}
 	}
 
+    protected void enable(bool isEnabled)
```

Comment:
This seems generic enough to justify moving to the superclass and overriding here